iati/core/resources/lib_data/validation_err_codes.yaml

```diff
@@ -233,3 +233,123 @@
         For more information about Rulesets, see http://iatistandard.org/rulesets/
     info: |-
         The way in which the Dataset does not conform is unknown.
+
+- err-rule-at-least-one-conformance-fail:
+    base_exception: ValueError
+    category: rule
+    description: |-
+        A Dataset does not conform with an 'atleast_one' Rule.
+    help: |-
+        The 'atleast_one' Rule specifies that there must be at least one element described by the given paths.
+        For more information about this Rule, see http://iatistandard.org/rulesets/
+    info: |-
+        The way in which the Dataset does not conform is unknown.
```

Comment:
Probably want to provide a string output of the Rule itself somewhere (so that you know which Rule of this type has failed).